src/Storages/AlterCommands.cpp

```diff
@@ -350,6 +358,24 @@ std::optional<AlterCommand> AlterCommand::parse(const ASTAlterCommand * command_
 }
 
 
+void AlterCommands::apply(DatabasePtr database, ContextPtr context) const
+{
+    for (const AlterCommand & command : *this)
+    {
+        if (!command.ignore)
+        {
+            if (command.type == AlterCommand::MODIFY_DATABASE_SETTING)
+            {
+                database->tryApplySettings(command.settings_changes, context);
```

Comment:
What if database will be dropped here?

Reply:
Changed the order of writing settings changes to disk and writing to postgresql replication flow.
Then if we crash in between, it will be Ok for this database engine because it always checks metadata on startup and makes according modifications. Though cannot say the same for other engines if they will also support settings changes..